### PR TITLE
Add node class

### PR DESCRIPTION
**Pull request by FokkeZB**
Includes updated tests

Replaces #135 

@johanstokking please review and merge once test pass (they will)

### CODE REVIEW

File: examples/Node/Node.ino [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+  ttn.sendBytes(payload, sizeof(payload), PORT_BUTTON);
+}
+
+void onMessage(const byte* payload, size_t length, port_t port) {
+  debugSerial.println("-- ON MESSAGE");
+
+  uint32_t interval;
+
+  switch (port) {
+    case 2:
```
Comment (johanstokking) [inline examples/Node/Node.ino]: Use a named port

File: examples/Node/Node.ino [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+      interval = ((payload[1] << 8) + payload[2]) * 1000;
+      node->configInterval(true, interval);
+      debugSerial.println("Loop delay changed: " + String(interval));
+      loop();
+      break;
+    case 3:
```
Comment (johanstokking) [inline examples/Node/Node.ino]: Use a named port

File: examples/Node/Node.ino [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  debugSerial.println("-- ON MESSAGE");
+
+  uint32_t interval;
+
+  switch (port) {
+    case 2:
+      interval = ((payload[1] << 8) + payload[2]) * 1000;
+      node->configInterval(true, interval);
+      debugSerial.println("Loop delay changed: " + String(interval));
+      loop();
```
Comment (johanstokking) [inline examples/Node/Node.ino]: Is this safe?

File: examples/Node/Node.ino [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+  bytes = (byte*) &temperature;
+  payload[5] = bytes[1];
+  payload[6] = bytes[0];
+
+  payload[7] = node->getColor();
+
+  byte flags = B00000000;
+  
+  if (node->isMoving()) {
+    flags = flags | 1;
```
Comment (johanstokking) [inline examples/Node/Node.ino]: `flags |= 1;`

**Comment by FokkeZB** [comment]
@johanstokking requests addressed

File: test/TheThingsNode/TheThingsNode.ino [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    bool moving = node->isMoving();
+    bool pressed = node->isButtonPressed();
+    ttn_color color = node->getColor();
+    String colorStr = node->colorToString(color);
+    bool red = node->getRed();
+    bool green = node->getGreen();
+    bool blue = node->getBlue();
+    bool connected = node->isUSBConnected();
+    uint16_t battery = node->getBattery();
+}
```
Comment (johanstokking) [inline test/TheThingsNode/TheThingsNode.ino]: Line ending

File: src/TheThingsNode.h [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  TTN_YELLOW,
+  TTN_CYAN,
+  TTN_MAGENTA,
+  TTN_WHITE,
+  TTN_BLACK
+};
+
+class TheThingsNode
+{
+  public:
```
Comment (johanstokking) [inline src/TheThingsNode.h]: Start with private declarations

**Comment by FokkeZB** [comment]
Addressed

**Pull request merged**